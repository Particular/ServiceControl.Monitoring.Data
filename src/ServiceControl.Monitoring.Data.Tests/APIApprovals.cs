#if NET452
namespace ServiceControl.Monitoring.Data.Tests
{
    using System.IO;
    using System.Runtime.CompilerServices;
    using ApprovalTests;
    using ApprovalTests.Core;
    using ApprovalTests.Writers;
    using NUnit.Framework;
    using PublicApiGenerator;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Approve()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(RingBuffer).Assembly);
            Approvals.Verify(WriterFactory.CreateTextWriter(publicApi, "txt"), GetNamer(), Approvals.GetReporter());
        }

        IApprovalNamer GetNamer([CallerFilePath] string path = "")
        {
            var dir = Path.GetDirectoryName(path);
            var name = typeof(RingBuffer).Assembly.GetName().Name;

            return new Namer
            {
                Name = name,
                SourcePath = dir,
            };
        }

        class Namer : IApprovalNamer
        {
            public string SourcePath { get; set; }
            public string Name { get; set; }
        }
    }
}
#endif