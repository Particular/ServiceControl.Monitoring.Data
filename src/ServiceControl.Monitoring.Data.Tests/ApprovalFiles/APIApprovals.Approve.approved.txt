[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("ServiceControl.Monitoring.Data.Tests")]